src/main/java/reactor/rabbitmq/Sender.java

```diff
@@ -161,7 +161,8 @@ protected Scheduler createScheduler(String name) {
                     // so we need to complete in another thread
                     channelCloseThreadPool.execute(() -> channelCloseHandler.accept(signalType, channel));
                 }
-            }));
+            }))
+            .publishOn(sendOptions.getScheduler(), sendOptions.getMaxInFlight());
```

Comment:
Why do you need to publish it on a different scheduler?

Reply:
I think `publishOn` is called only to benefit from the prefetch it provides. I don't know any other ways to do so without using a scheduler. Maybe a reasonable default for the scheduler would then be `Schedulers#immediate`, to avoid changing the threading context.

Reply:
yes, and that's exactly what "limitRate" does under the covers ;)